### ISSUE TITLE
BoolQuery with no queries inside will generate invalid ES query

### ISSUE THREAD

**Issue by pmishev** [ongr-io/ElasticsearchDSL#260]
If there are no queries added to a BoolQuery, it will generate:
```
  "query": {
    "bool": []
  },
```
and the correct output should be:
```
  "query": {
    "bool": {}
  },
```
So you should add a check to see if there are no queries and then return new \stdObject instead of an empty array.

**Comment by saimaz** [comment]
yes, that's correct, thank's for the notice.

**Comment by saimaz** [comment]
It is released in 5.0.4, in a day or to it will be released also in the upcoming 6.0.2

**Comment by dannydorfel** [comment]
Just trying to help by fixing the easiest tasks @saimaz .
This issue should be closed, but still seems to be open in GitHub ;-)
